mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -657,21 +698,25 @@ def _explain_statement(self, cursor, statement, schema, obfuscated_statement):
         Tries the available methods used to explain a statement for the given schema. If a non-retryable
         error occurs (such as a permissions error), then statements executed under the schema will be
         disallowed in future attempts.
+        returns: An execution plan, otherwise it returns a list of tuples containing: (strategy, error code, error)
+        rtype: Optional[Dict], List[Tuple[Optional[str], Optional[DBExplainError], Optional[Exception]]]
         """
         start_time = time.time()
         strategy_cache_key = 'explain_strategy:%s' % schema
-        explain_strategy_error = 'ERROR'
         tags = self._tags + ["schema:{}".format(schema)]
 
         self._log.debug('explaining statement. schema=%s, statement="%s"', schema, statement)
 
         if not self._can_explain(obfuscated_statement):
             self._log.debug('Skipping statement which cannot be explained: %s', obfuscated_statement)
-            return None
+            return None, [(None, DBExplainError.no_plans_possible, None)]
 
-        if self._collection_strategy_cache.get(strategy_cache_key) == explain_strategy_error:
-            self._log.debug('Skipping statement due to cached collection failure: %s', obfuscated_statement)
-            return None
+        strategy_cache = self._collection_strategy_cache.get(strategy_cache_key)
```

Comment:
This should be named `cached_strategy`. The current name implies that it's a cache.